module/documents/actor/actor.mjs

```diff
@@ -3254,28 +3278,42 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
   /**
    * Flash ring & display changes to health as scrolling combat text.
-   * @param {object} options  Any data provided by the update.
+   * @param {object} changes          Object of changes to hit points.
+   * @param {number} changes.hp       Changes to `hp.value`.
+   * @param {number} changes.temp     The change to `hp.temp`.
+   * @param {number} changes.total    The total change to hit points.
    * @protected
    */
-  _displayTokenEffect(options) {
-    const dhp = Number(options.dhp);
+  _displayTokenEffect(changes) {
+    let key;
+    let change;
+    if ( !parseInt(changes.hp) && parseInt(changes.temp) ) key = "temp";
+    else {
+      key = "hp";
+      change = changes.total;
+    }
+    if ( !key ) return;
+    change ??= changes[key];
+
     const tokens = this.isToken ? [this.token] : this.getActiveTokens(true, true);
+    if ( !tokens.length ) return;
+
+    const pct = Math.clamped(Math.abs(change) / this.system.attributes.hp.max, 0, 1);
+    const fill = CONFIG.DND5E.tokenHPColors[(key === "hp") ? (change < 0 ? "damage" : "healing") : "temp"];
+
     for ( const token of tokens ) {
       if ( !token.object.visible || !token.object.renderable ) continue;
-      token.flashRing(options);
-      if ( dhp ) {
-        const t = token.object;
-        const pct = Math.clamped(Math.abs(dhp) / this.system.attributes.hp.max, 0, 1);
-        canvas.interface.createScrollingText(t.center, dhp.signedString(), {
-          anchor: CONST.TEXT_ANCHOR_POINTS.TOP,
-          // Adapt the font size relative to the Actor's HP total to emphasize more significant blows
-          fontSize: 16 + (32 * pct), // Range between [16, 48]
-          fill: CONFIG.DND5E.tokenHPColors[dhp < 0 ? "damage" : "healing"],
-          stroke: 0x000000,
-          strokeThickness: 4,
-          jitter: 0.25
-        });
-      }
+      token.flashRing({ [key]: change });
```

Comment:
It seems `flashRing` still expects `dhp` or `dtemp` arguments.

Reply:
I considered it less breaking to leave the arguments for this method as is, and only getting rid of the update options. I suspect those would have been much less used (they were never quite reliable if one wanted to use those to check for hp changes).

One thing we could do is change it to a single argument here that is just the string "temp", "damage", or "healing", but lemme know your thoughts on that.

Reply:
I'm sorry, I'm not following. What update options? The issue was that this line, as-written, was doing nothing because it was passing the incorrect arguments to `flashRing`.

Reply:
Oh I see what you mean now, yeah that must have been an oversight. Regardless perhaps the change I made makes the function's use a bit clearer?